src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -136,24 +88,57 @@ class WorkspaceDashboardContent extends Component {
     }
   }
 
+  async save() {
+    const { refreshWorkspace, workspace: { workspace: { namespace, name } }, ajax: { Workspaces } } = this.props
+    const { editDescription: description } = this.state
+    try {
+      this.setState({ saving: true })
+      await Workspaces.workspace(namespace, name).shallowMergeNewAttributes({ description })
+      refreshWorkspace()
+    } catch (error) {
+      reportError('Error saving workspace', error)
+    } finally {
+      this.setState({ saving: false, editingWorkspace: false })
+    }
+  }
+
   render() {
-    const { workspace, refreshWorkspace, workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
-    const { submissionsCount, storageCostEstimate, editingWorkspace } = this.state
+    const { workspace: { accessLevel, workspace: { createdDate, lastModified, bucketName, attributes: { description } } } } = this.props
+    const { submissionsCount, storageCostEstimate, editingWorkspace, editDescription = '', saving } = this.state
```

Comment:
This isn't really "editing workspace" any more, right? Also, we should be able to track both whether we're editing and the edited text as a single variable.